include/seqan3/core/detail/strong_type.hpp

```diff
@@ -64,12 +65,65 @@ constexpr bool add_enum_bitwise_operators<seqan3::detail::strong_type_skill> = t
 namespace seqan3::detail
 {
 
+/*!\interface seqan3::detail::strong_type_specialisation <>
+ * \brief Defines the requirements of a seqan::detail::strong_type specialisation.
+ * \tparam strong_type_t The type the concept check is performed on (the putative strong type).
+ * \ingroup core
+ */
+/*!\name Requirements for seqan3::detail::strong_type_specialisation
+ * \brief You can expect these members on all types that implement seqan3::detail::strong_type_specialisation.
+ * \relates seqan3::detail::strong_type_specialisation
+ * \{
+ */
+//!\cond
+template <typename strong_type_t>
+SEQAN3_CONCEPT strong_type_specialisation = requires (strong_type_t && obj)
+{
+//!\endcond
+
```

Comment:
Stupid question, but wouldn't something like this suffice?

```c++
template <typename strong_type_t>
SEQAN3_CONCEPT strong_type_specialisation = requires (strong_type_t && obj)
{
    typename strong_type_t::value_type;
    {strong_type_t::skills};
    requires std::derived_from<strong_type_t, // I forgot the order of arguments.
                               strong_type<typename strong_type_t::value_type,
                                           strong_type_t,
                                           strong_type_t::skills>>;
}
```

with some additional `std::remove_cvref_t` for `strong_type_t &` instances?

I think this would be a bit more close to the intention of type specialisation. (I forgot in our earlier discussion that the strong type is inherited, so `is_type_specialisation_of_v` wouldn't have worked even without mixed template arguments.)

Reply:
did you check?

Reply:
still open?

Reply:
no